minkindr/include/kindr/minimal/rotation-quaternion.h

```diff
@@ -156,9 +156,18 @@ class RotationQuaternionTemplate {
     return vector() == rhs.vector();
   }
 
+  /// \brief Check the validity of a rotation matrix.
+  static bool isValidRotationMatrix(const RotationMatrix& matrix);
+
+  /// \brief Factory to construct a RotationQuaternionTemplate from a near
+  ///        orthonormal rotation matrix.
+  inline static RotationQuaternionTemplate<Scalar> constructAndRenormalize(
+      const RotationMatrix& R) {
+    return RotationQuaternionTemplate<Scalar>(Implementation(R).normalized());
```

Comment:
But how is then defined what this function actually does? Don't we want to get the quaternion corresponding to the proper rotation matrix closest to R? I strongly doubt that Eigen would provide that without any further requesting it. As finding the closest SO(3) matrix isn't a very cheap operation. But would not the user of constructAndRenormalize expect that?

If yes:
@kruesip and me recently thought about that and found a proper solution using the SVD implementation of Eigen. We could at least uses this concept to evaluate the here suggested simple version.